File: Python 3/EquiIndex.py
def solution(array):
    size = len(array)
    if size == 0:
        return -1
    sum = 0
    before = []
    for i in range(size):
        sum = sum + array[i]
        before.append(sum)
    if (sum - array[0]) == 0:
        return 0
    for i in range(1, size):
        res = sum - (array[i] + before[i-1])
        if res == before[i-1]:
            return i
    return -1

File: C/EquiIndex.c
int equiIndex(int array[], int size) {
    long before[size];
    long sum = 0;
    for(int i = 0; i<size; ++i){
        sum += array[i];
        before[i] = sum;
    }
	if(!size)
		return -1;
    if((sum - array[0]) == 0)
        return 0;
    for(int i = 1; i<size; ++i){
        int res = sum - (array[i] + before[i-1]);
        if(res == before[i-1])
            return i;
    }
    return -1;
}